### DISCUSSION TITLE
How to get outputs from REGEX VARIABLE block

### DISCUSSION THREAD

**Discussion by loic-riviere** [AutomaApp/automa#1252] [Q&A]
I attempt to extract the domain name from the active tab URL. I try and use a _RegEx Variable_ block but cannot figure out how to get results.

For instance, the URL **https://docs.automa.site/workflow/expressions.html**  => I expect it returns  **https://docs.automa.site/**

URL is in a variable from a _Get Tab URL_ block : source_url.

In a _RegEx Variable_ block, I specify:
- variable: source_url
- method: match value
- regex: my regex (/^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n\?\=]+)/im)

I want to get the matching result to open it in a new tab. But I cannot find how to get it.  Documentation does not specify output. I inspected variables in debug mode, in vain.

Does anyone knows how to do it ?

**Comment by fmen1** [comment]
Under the Form Block you can place  {{variables.source_url}} in under the "text field" option.

**Reply by loic-riviere** [reply]
Indeed, I could. But it just displays the source_url variable. The result will have nothing to do with the regex operation.

To be more specific, I want to get the part of the variable that matches the regex. And to use it in an "Open tab" block as the tab url. Any hint ?

**Comment by fmen1** [comment]
I am by no stretch of the imagination an Automa expert, however.
assuming your regex works (it did not work for me in regex101). here's how I do it....

------------------------------------------------------------------------------------------
Variable name
 source_url
Method

Match value
RegEx
/(/^(?:https?://)?(?:[^@\n]+@)?(?:www.)?([^:\/\n\?\=]+)/im)            /gi
------------------------------------------------------------------------------------------

Do a trial to see if it works with a Data Export Block
-------------------------------------------------------------------------------------------
Data to export

Variable
source_url
File name
url
On conflict

Overwrite
Export as

Plain text
---------------------------------------------------------------------
Remember that the Form Block requires the following format for the variable: {{variables.source_url}} 

Good Luck!
